File: leetcode/twoSum.cpp
class Solution {
public:
    // vector<int> twoSum(vector<int>& nums, int target) {
    //     int index1,index2;
    //     vector<int> result;
    //     index1=index2 = 0;
    //     for(int i=0;i<nums.size();i++){
    //         for(int j=i+1;j<nums.size();j++){
    //             if(nums[i]+nums[j]==target){
    //                 // index1 = i;
    //                 // index2 = j;
    //                 result.push_back(i);
    //                 result.push_back(j);
    //                 break;
    //             }
    //         }
    //     }
    //     return result;
    // }
        vector<int> twoSum(vector<int>& nums, int target) {
        map<int,int> m;
            
        vector<int> result;
        for(int i=0;i<nums.size();i++){
            m[nums[i]] = i;// key is 数字；
        }
            
        for(int i=0;i<nums.size();i++){
           int complement = target - nums[i];
            if(m.count(complement) && m[complement] != i){
                result.push_back(i);
                result.push_back(m[complement]);
                break;
                }
            }
  
        return result;
    }
    
};

File: leetcode/two-sum.py
'''
Author: Ful Chou
Date: 2021-01-03 00:29:28
LastEditors: Ful Chou
LastEditTime: 2021-01-03 00:30:13
FilePath: /leetcode/two-sum.py
Description: What this document does

简单题不多说，空间换时间 链接： https://leetcode-cn.com/problems/two-sum/
'''
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # from collections import defaultdict
        '''
        哈希表 加快 find 速度
        '''
        dic = {} 
        for idx,num in enumerate(nums):
            dic[num] = idx
        for idx,num in enumerate(nums):
            res = target - num
            if res in dic and dic[res] != idx:
                return [idx, dic[res]]
        return [-1,-1]
